### PR TITLE
Add checkBasicAuth function.

### LINKED ISSUE #1449: performBasicAuth: plz separate "Authorization" header read code and other stuff
Two performBasicAuth functions contain code duplication and also without separate header read function it isn't possible to check auth data without different response cases.

(Faced out with it because I am writing server what uses auth info as parameter, but response isn't depends from availability of auth info.)

### PR DESCRIPTION
**Pull request by s-ludwig**
Separates the authorization test from authorization enforcement.

Fixes #1449.

### CODE REVIEW

**Comment by s-ludwig** [comment]
@denizzzka: Is `checkBasicAuth` sufficient for your use case, or is more control necessary?

File: http/vibe/http/auth/basic_auth.d [hunk 1]
```diff
@@ -71,28 +55,54 @@ HTTPServerRequestDelegateS performBasicAuth(string realm, bool delegate(string,
...
+		pwcheck = A delegate queried for validating user/password pairs
+
+	Returns: Returns `true` $(I iff) a valid Basic Auth header is present
+		and the credentials were verified successfully by the validation
+		callback.
+
+	Throws: Throws a `HTTPStatusExeption` with `HTTPStatusCode.badRequest`
+		if the "Authorization" header is malformed.
+*/
+bool checkBasicAuth(scope HTTPServerRequest req, scope PasswordVerifyCallback pwcheck)
```
Comment (denizzzka) [inline http/vibe/http/auth/basic_auth.d]: Can ``req`` argument be a ``const`` (or ``in``) type?

**Comment by denizzzka** [comment]
@s-ludwig checkBasicAuth exactly what is needed. Thanks!

(I am changed my code with checkBasicAuth, but can't run my integration tests because something changed in vibe.d: #1688.)

**Comment by s-ludwig** [comment]
Okay, I think it should be safe to still merge this into 0.8.0.

Regarding const/in, I don't want to shut the door for more lazy evaluation and reference counting, which is why the `HTTPServerRe*` values are always passed in plain form.

**Pull request merged**